### PR TITLE
Fix crash in adaptive convection when used with baseboard heat and lights with return air heat gain

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Addresses #6391 where adaptive convection would crash during zone sizing when used with baseboard heat (or similar non-forced-air zone equipment) and lights with return air heat gain.

New unit test in [ConvectionCoefficients.unit.cc](https://github.com/NREL/EnergyPlus/blob/1b2270515856f18b98a74bc6d077cfddb96e2a1c/tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc#L108) reproduces the crash at [this line in ConvectionCoefficients::DynamicIntConvSurfaceClassification](https://github.com/NREL/EnergyPlus/blob/1f84213a71d31a2094a639730a04cec4f1eae841/src/EnergyPlus/ConvectionCoefficients.cc#L6521) due to accessing an unallocated array `SNLoadPredictedRate`.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by mjwitte** [comment]
Crash is fixed, no other diffs, and unit test coverage increased by 148 lines - woohoo! Ready for review.

File: tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc [hunk 1]
```diff
@@ -99,4 +108,454 @@ TEST_F( EnergyPlusFixture, ConvectionCoefficientsTest_ConvectionCofficients )
...
+	DynamicIntConvSurfaceClassification( 13 ) ; 
+	EXPECT_EQ( DataSurfaces::Surface( 13 ).IntConvClassification, DataSurfaces::InConvClass_A3_StableTilted );
+
+	DynamicIntConvSurfaceClassification( 14 ) ; 
+	EXPECT_EQ( DataSurfaces::Surface( 14 ).IntConvClassification, DataSurfaces::InConvClass_A3_StableHoriz );
+
+	DynamicIntConvSurfaceClassification( 15 ) ; 
+	EXPECT_EQ( DataSurfaces::Surface( 15 ).IntConvClassification, DataSurfaces::InConvClass_A3_StableHoriz );
+
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc:560]: Thorough!! :+1:

File: tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc [hunk 2]
```diff
@@ -99,4 +108,454 @@ TEST_F( EnergyPlusFixture, ConvectionCoefficientsTest_ConvectionCofficients )
...
+	HeatBalanceManager::GetConstructData( errorsFound ); // read construction data
+	EXPECT_FALSE( errorsFound ); // expect no errors
+
+	HeatBalanceManager::GetZoneData( errorsFound );
+	ASSERT_FALSE( errorsFound );
+
+	SurfaceGeometry::SetupZoneGeometry( errorsFound );
+	ASSERT_FALSE( errorsFound );
+	HeatBalanceManager::AllocateHeatBalArrays();
+	HeatBalanceSurfaceManager::AllocateSurfaceHeatBalArrays();
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc:430]: I started feeling like this was a lot of setup needed for this, but it actually makes great sense.  I like this.

File: src/EnergyPlus/ConvectionCoefficients.cc [hunk 3]
```diff
@@ -6394,7 +6394,7 @@ namespace ConvectionCoefficients {
 			FlowRegimeStack( 0 ) = InConvFlowRegime_A3;
 		} else { // is controlled, lets see by how and if that means is currently active
 
-			if ( ! ( ZoneEquipConfig( ZoneNum ).EquipListIndex > 0 ) ) {
+			if ( ! ( ZoneEquipConfig( ZoneNum ).EquipListIndex > 0 ) || SysSizingCalc || ZoneSizingCalc || ! ZoneEquipSimulatedOnce ) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/ConvectionCoefficients.cc:6397]: Makes sense to me.  As for whitespace, I don't mind if the ! is right up against the boolean, like !something.  But no worries here.

**Review by Myoldmopar — APPROVED** [review]

**Comment by EnergyBloch** [comment]
Sorry, but how do I make these changes on my laptop? 

I cant seem to find the file(s) that i need to edit?

**Comment by mjwitte** [comment]
@EnergyBloch Instructions to build EnergyPlus locally may be found [here](https://github.com/NREL/EnergyPlus/wiki/BuildingEnergyPlus). If you want to build v8.8.0 plus this fix, then checkout the develop branch at the v8.8.0 tag, build that, then add this fix by hand (unless you are a git wizard and know how to merge in this specific commit).

**Pull request merged**